---
title: "Compare FBXO11 transcriptome and proteomics results"
author: "Jennifer Grants"
date: "2/4/2019"
output: 
  html_document:
    keep_md: true
    toc: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, eval = TRUE)
library(tidyverse)
```

# Predictions  

* Proteins that are ubiquitination targets may be detected as FBXO11 binding targets.  

* The network of FBXO11 protein targets may share common functions with the network of genes upregulated in shFBXO11 cells. i.e. Loss of FBXO11 produces gene expression changes due to increased abundance of FBXO11 protein targets.  



# Ubiquitination/binding overlap
```{r}
load("./FBXO11/Proteomics/rda/ub.target.list.rda")
load("./FBXO11/Proteomics/rda/binding.target.rda")
```

```{r}
overlap <- intersect(ub.target.list, binding.target)
overlap
```

> Conclusion: There is no overlap between the Ub target list and the binding target list. They are very small lists, and Ub interactions could be too short-lived/weak to be detected in the coIP.  


# Prepare proteomics lists for STRING analysis  
```{r}
ub.targets <- data.frame(Target = ub.target.list)
write.csv(x = ub.targets, file = "./FBXO11/Proteomics/Results_summaries/diGly_result_ub_targets.csv", row.names = F)

bd.targets <- data.frame(Binding = binding.target)
write.csv(x = bd.targets, file = "./FBXO11/Proteomics/Results_summaries/coIP_result_binding_partners.csv", row.names = F)
```

> Performed STRING analysis on protein-protein interactions, then Autoannotate based on MCL clusters analysis (4 Biggest Words) of STRING "description".

*  See: `STRING_onProteomics` folder.

# Prepare GSEA GO-terms "Immune" results Leading Edge genes for STRING analysis
Use gitbash to move all gene lists with "IMMUNE" in GO term name to a new folder, `Immune_results`.  
Then commented out because I only need to do it once.  
```{bash}
#cd FBXO11/GSEA_onLimma/FBXO11_commonEffectOnly_GO.Gsea.1549051321021
#ls

#mkdir ./Immune_results

#mv GO*IMMUNE*.xls ./Immune_results

#cd ../../../
#pwd
```

Manually these re-save GSEA results as .csv, because GSEA creates .xls files that are not actually .xls files.  
Then move the .xls files to a different folder and keep the .csv files in same folder. 
Commented out because I only need to run once.  
```{bash}
#cd FBXO11/GSEA_onLimma/FBXO11_commonEffectOnly_GO.Gsea.1549051321021/Immune_results
#mkdir xls_files
#mv *.xls ./xls_files/
```


Read in .csv files from `Immune_results` folder,  filter to Leading Edge genes only, and select gene name column:  
```{r warning=FALSE, message=FALSE}
library(readxl)

files <- list.files("FBXO11/GSEA_onLimma/FBXO11_commonEffectOnly_GO.Gsea.1549051321021/Immune_results", full.names = T)
files

for(i in 1:(length(files)-1)){
  df <- read_csv(files[i]) %>%
    filter(`CORE ENRICHMENT` == "Yes") %>%
    select(`PROBE`)
  assign(paste0("d", i), df)
}

rm(df)

```


## Compile list of all leading edge genes (unique)
```{r}
le <- rbind(d1, d2, d3, d4, d5) %>% unlist() %>% as.character()

length(le)
sum(nrow(d1), nrow(d2), nrow(d3), nrow(d4), nrow(d5))

le <- unique(le)
length(le)
```

Save results to folder `STRING_onImmuneLE`:  
```{r}
write.csv(le, "./FBXO11/STRING_onImmuneLE/GO_Immune_leadingEdge_unique.csv", row.names = F)
```

