---
title: "PMP Cell Line Data Exploration"
author: "Jennifer Grants"
date: "1/8/2019"
output:
  html_document:
    keep_md: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, eval = TRUE)
```

```{r, include = FALSE}
library(tidyverse)
library(amlpmpdata)
library(knitr)
```



## Load specimen list and identify cell lines samples of interest
```{r load_specimens}
specimens <- amlpmpdata::db_specimens

```

Add patient metadata, and then filter out patient samples:
```{r add_pt_meta}
pt_meta <- amlpmpdata::db_patients

specimens$is_patient <- pt_meta[match(specimens$patient_external_id, pt_meta$external_id),]$is_patient

kable(head(specimens))
```



```{r filter_out_pt}
cells <- specimens[which(specimens$is_patient == F),]

kable(head(cells))
```

```{r}
write.csv(cells, "./output/summary_non_patient_samples.csv", row.names = F)
```


## Investigate MDS-L experiments 
> Jeremy's suggestion of most useful to our lab
> Focus on triplicate experiment: MDS-L + DMSO vs. Len & MDS-L-Runx1-KO + DMSO vs. Len

```{r}
int <- filter(cells, patient_external_id == "MDS-L" & grepl(x = cells$meta, pattern = "human"))

kable(int)
```

> Hypothesis:
* Loss of Runx1 reduces Lenalidomide (Len) sensitivity of MDS-L cell line by altering the gene expression response to Len.

> Aims:
* To identify Len-responsive genes: i.e. differentially expressed DMSO vs. Len in parental MDS-L cell line.
* To identify Runx1-dependent genes: i.e. differentially expressed in Len-treated MDS-L vs. KO (perhaps normalize to DMSO condition in both cell lines?)

