---
title: "PMP Cell Line Data Exploration"
author: "Jennifer Grants"
date: "1/8/2019"
output:
  html_document:
    keep_md: true
    toc: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, eval = TRUE)
```

```{r, include = FALSE}
library(tidyverse)
library(amlpmpdata)
library(knitr)
```



## Load specimen list and identify cell lines samples of interest
```{r load_specimens}
specimens <- amlpmpdata::db_specimens

```

Add patient metadata, and then filter out patient samples:
```{r add_pt_meta}
pt_meta <- amlpmpdata::db_patients

specimens$is_patient <- pt_meta[match(specimens$patient_external_id, pt_meta$external_id),]$is_patient

kable(head(specimens))
```



```{r filter_out_pt}
cells <- specimens[which(specimens$is_patient == F),]

kable(head(cells))
```

```{r summary_non_pt}
write.csv(cells, "./output/summary_non_patient_samples.csv", row.names = F)
```


## Investigate MDS-L experiments 
> Jeremy's suggestion of most useful to our lab  
> Focus on triplicate experiment: MDS-L + DMSO vs. Len & MDS-L-Runx1-KO + DMSO vs. Len

```{r mdsl_samples}
int <- filter(cells, patient_external_id == "MDS-L" & grepl(x = cells$meta, pattern = "human"))

kable(int)
```

```{r summary_mdsl}
write.csv(int, "./output/summary_MDSL_triplicate.csv", row.names = F)
```


> Hypothesis:  

* Loss of Runx1 reduces Lenalidomide (Len) sensitivity of MDS-L cell line by altering the gene expression response to Len.

> Aims:  

* To identify Len-responsive genes: i.e. differentially expressed DMSO vs. Len in parental MDS-L cell line.  
* To identify Runx1-dependent genes: i.e. differentially expressed in Len-treated MDS-L vs. KO (perhaps normalize to DMSO condition in both cell lines?)


## Access expression data for samples of interest
```{r find_libs}
libs <- amlpmpdata::db_libraries[match(int$specimen_external_id, amlpmpdata::db_libraries$specimen_subset_external_id),]
```

How to retrieve expr data tables:
We can find all mRNA expression analysis files by querying the db_paths table. This gives a separate file for each sample. MShadbolt wrote a python script to join all these files and create one big tsv containing **tpm counts** that can be loaded as follows:

file://isaac/klabanalysis/mshadbolt/KARSANBIO-1352_Karsan_Lab_Informatics_Support/KARSANBIO-1521_mRNA_and_miRNA_data_retrieval/report/KARSANBIO-1521_mRNA_and_miRNA_data_retrieval.html

```{r load_mrna}
mrnas <- read_tsv("/projects/karsanlab/PMP-AML/expression/bcbio_fastrnaseq_salmon/salmon_AML_PMP_expression_output.tsv")
```

```{r prev_mrna}
kable(mrnas[1:5, 1:5])
```


#### Select only libraries of interest
```{r libs_of_int_mrna}
Name <- mrnas$Name
expr <- mrnas[, match(libs$library_name, colnames(mrnas))]
expr$Name <- Name
expr <- select(expr, Name, everything())

kable(head(expr))
```



## Save Rdata files for downstream analysis
```{r}
save(expr, file = "expr.RData")
save(libs, file = "libs.RData")
```



