---
title: "PMP Cell Line Data Exploration"
author: "Jennifer Grants"
date: "1/8/2019"
output:
  html_document:
    keep_md: true
    toc: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, eval = TRUE)
```

```{r, include = FALSE}
library(tidyverse)
library(amlpmpdata)
library(knitr)
```



## Load specimen list and identify cell lines samples of interest
```{r load_specimens}
specimens <- amlpmpdata::db_specimens

```

Add patient metadata, and then filter out patient samples:
```{r add_pt_meta}
pt_meta <- amlpmpdata::db_patients

specimens$is_patient <- pt_meta[match(specimens$patient_external_id, pt_meta$external_id),]$is_patient

kable(head(specimens))
```



```{r filter_out_pt}
cells <- specimens[which(specimens$is_patient == F),]

kable(head(cells))
```

```{r summary_non_pt}
write.csv(cells, "./output/summary_non_patient_samples.csv", row.names = F)
```


## Investigate datasets of interest  

## Kate's miR SPG experiment  

```{r kate_samples}
int <- filter(cells, patient_external_id == "UT-7")

kable(int)
```

```{r summary_Kate}
write.csv(int, "./Kate/summary.csv", row.names = F)
```


> Experimental design: Kate's data  

* Original sponge (OR) that targets miR-143 and miR-X  
* miR-143-specific sponge (143fix)  
* miR-X-specific sponge (Xfix)  
* vs. GFP control or untransduced UT-7 cells

> Rationale: (Kate's thesis)  

* The original miR-143 sponge contains several potential repetitive seed sequences, besides that of miR-143 (seed sequence GAGAUGA), and so may target additional miR's  
* Among these putative miR seeds, Kate identified a complementary novel miR (miR-X, seed sequence CUGUAGC) that is: 
    * expressed in human AML  
    * predicted to target several genes whose protein products were upregulated in a SILAC experiment that used the OR sponge (greater Obs/Pred ratio of target upregulation than miR-143; still very low though)  

> Aims:  

* To identify miR-143-specific and miR-X-specific target genes  
    * based on mRNA upregulation upon specific fixed sponge treatments vs. OR sponge treatment
* To compare miR-143 and miR-X targets to Kate's SILAC results  


### Access expression data for samples of interest  
```{r find_libs_Kate}
libs <- amlpmpdata::db_libraries[match(int$specimen_external_id, amlpmpdata::db_libraries$specimen_subset_external_id),]
```

How to retrieve expr data tables:  
We can find all mRNA expression analysis files by querying the db_paths table. This gives a separate file for each sample. MShadbolt wrote a python script to join all these files and create one big tsv containing **tpm counts** that can be loaded as follows:  

file://isaac/klabanalysis/mshadbolt/KARSANBIO-1352_Karsan_Lab_Informatics_Support/KARSANBIO-1521_mRNA_and_miRNA_data_retrieval/report/KARSANBIO-1521_mRNA_and_miRNA_data_retrieval.html  

```{r load_mrna}
mrnas <- read_tsv("/projects/karsanlab/PMP-AML/expression/bcbio_fastrnaseq_salmon/salmon_AML_PMP_expression_output.tsv")
```


### Select only libraries of interest  
```{r libs_of_int_Kate}
Name <- mrnas$Name
expr <- mrnas[,which(colnames(mrnas) %in% libs$library_name)]
expr$Name <- Name
expr <- select(expr, Name, everything())

kable(head(expr))
```


### Save Rdata files for downstream analysis  
```{r save_Kate}
save(expr, file = "./Kate/expr.RData")
save(libs, file = "./Kate/libs.RData")
```



## FBXO11 knockdown  
```{r fbxo11_samples}
int <- filter(cells, grepl(pattern = "Linda", x = cells$meta))

kable(int)
```


```{r summary_FBXO11}
write.csv(int, "./FBXO11/summary.csv", row.names = F)
```


> Experimental design: FBXO11 knockdown data  

* FBXO11 shRNA knockdown (2 different shRNA constructs)  
* vs. sh-Scrambled controls  
* (in triplicate)  


> Rationale and Hypothesis:  

* FBXO11 is part of a E3 ubiquitin ligase complex, therefore loss of FBXO11 will affect protein expression level. These changes may in turn affect gene expression, causing changes in cellular function.  
* FBXO11 is lost in some AMLs, and FBXO11 loss in mice causes AML-like disease when combined with a predisposing mutation (AML-ETO). Therefore the functional changes evoked by FBXO11 knockdown may be causal in AML development.  
* _Hypothesis_: Loss of FBXO11 causes gene expression changes that alter cellular function to promote AML development.  


> Aims:  

* To identify processes and pathways upregulated upon knockdown of FBXO11 (changes in cellular function)  
* To compare these results to proteomics data from Linda and Gerben to determine potential FBXO11-target-protein regulators of these gene expression changes  


### Access expression data for samples of interest  
```{r find_libs_FBXO11}
libs <- amlpmpdata::db_libraries[match(int$specimen_external_id, amlpmpdata::db_libraries$specimen_subset_external_id),]
```


### Select only libraries of interest  
```{r libs_of_int_FBXO11}
Name <- mrnas$Name
expr <- mrnas[,which(colnames(mrnas) %in% libs$library_name)]
expr$Name <- Name
expr <- select(expr, Name, everything())

kable(head(expr))
```


### Save Rdata files for downstream analysis  
```{r save_FBXO11}
save(expr, file = "./FBXO11/expr.RData")
save(libs, file = "./FBXO11/libs.RData")
```


## TIRAP overexpression (mouse BM)
```{r see_Rawa_samples}
tirap <- filter(cells, grepl(pattern = "Rawa", x = cells$meta))

kable(tirap)
```


### Use amlpmpdata package to identify path to files
```{r file_paths}
# merge library names to tirap table
libraries <- amlpmpdata::db_libraries

tirap$library <- libraries[match(tirap$specimen_external_id, libraries$specimen_subset_external_id),]$library_name

# merge path to tirap table  
paths <- amlpmpdata::db_paths

tirap$path <- paths[match(tirap$library, paths$library),]$path

kable(tirap)
```

> Paths are not available in this data package. Therefore I searched JIRA tickets and found Jenny's RPKM data for these samples, and saved to `TIRAP` / `data`.

* JIRA ticket: https://www.bcgsc.ca/jira/browse/KARSANBIO-412

### Load Jenny's pre-calculated RPKM values
```{r load_RPKM}
expr <- read_tsv(file = "./TIRAP/data/RPKM.tsv", col_names = T)
dim(expr) # should be 38215 rows according to RPKM.tsv

# fix column names for samples
cols <- data.frame(cols_names = colnames(expr[,9:ncol(expr)])) %>%
  separate(col = cols_names, into = "name", remove = T)
cols2 <- data.frame(name = colnames(expr[,1:8]))
cols3 <- rbind(cols2, cols)

colnames(expr) <- cols3$name

# get expression matrix only
expr <- dplyr::select(expr, gene_id, gene_symbol, contains("A54"))
save(expr, file = "./TIRAP/expr.rda")
```

### Save metadata
```{r save_meta}
libs <- amlpmpdata::db_libraries[match(tirap$specimen_external_id, amlpmpdata::db_libraries$specimen_subset_external_id),] 
save(libs, file = "./TIRAP/libs.rda")
```


> Experimental design: TIRAP overexpression data

* Bone marrow transfected with TIRAP overexpression vector or MIG empty vector, transplanted to n = 3 recipient mice
* Compare gene expression in TIRAP overexpression vs. empty vector

> Hypothesis:

* TIRAP overexpression causes gene expression changes that account for the bone marrow failure phenotype observed in these mice

> Aims:

* To identify differentially expressed genes in TIRAP-oe vs. Control (Limma)
* To identify up- and downregulated pathways and processes in TIRAP-oe vs. Control (GSEA, EnrichmentMap)
