---
title: "Expression analysis for Kate's RNA-seq data"
author: "Jennifer Grants"
date: "1/9/2019"
output:
  html_document:
    keep_md: true
    toc: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, eval = TRUE)
```

```{r include=FALSE}
library(tidyverse)
library(knitr)
library(pheatmap)
library(limma)
library(edgeR)
```

# Data and design  
`expr` is mRNA-seq TPM values, `libs` is metadata
```{r load_saved}
load("./Kate/expr.RData")
load("./Kate/libs.RData")
```

```{r add_design_factors}
libs$SPG_target <- ifelse(test = grepl(pattern = "GFP", x = libs$specimen_subset_external_id), yes = "ctrl", no = 
                                       ifelse(test = grepl(pattern = "OR", x = libs$specimen_subset_external_id), yes = "both", no =
                                                             ifelse(test = grepl(pattern = "143fix", x = libs$specimen_subset_external_id), yes = "miR143", no = 
                                                                      ifelse(test = grepl(pattern = "Xfix", x = libs$specimen_subset_external_id), yes = "miRX", no = "none"))))
```


```{r prev_libs}
kable(libs)
```

## Quality control checks  

### First convert row names from column "Names":  
```{r convert_row_names}
expr2 <- column_to_rownames(expr, var = "Name")

head(expr2)
```


### Dimensions:   
```{r dims}
# expression data
dim(expr2)

# metadata
dim(libs)
```


### NA values:   
```{r na_vals}
any(is.na(expr2))
```


### Range:  
```{r}
range(expr2)
```

```{r}
hist(as.matrix(expr2))
```
Appears not to be log transformed.    


## Data transformation / row normalization  
### Log2 transformation  
```{r log_transf}
# log2 transformation with constant +0.01 added to allow transformation of zeros
expr.log <- log2(expr2 + 0.01)

# Visualize log transformed data
hist(as.matrix(expr.log))

```
ASK: Why does this have so many 0 values? Is this normal for RNA-seq?  


# Limma
## Prep
### Check order of sample names and colnames in meta/expr
```{r}
order <- libs$library_name
expr2 <- expr2[,order]

identical(libs$library_name, colnames(expr2))
```

### Make model matrix from metadata
```{r limma_model_matrix}
libs$SPG_target <- factor(libs$SPG_target, levels = c("ctrl", "miR143", "miRX", "both", "none"))
designMatrix <- model.matrix(~SPG_target, libs)

head(designMatrix)
```

### Normalization and log transformation
```{r norm_log_transf}
# convert to matrix
expr.matrix <- column_to_rownames(expr, var = "Name") %>%
  as.matrix()

# filter out lowly expressed genes (keep: TPM must be >=1 in at least 2 samples)
thresh <- expr.matrix >= 1
keep <- rowSums(thresh) >=2

expr.matrix.keep <- expr.matrix[keep,]
dim(expr.matrix.keep)

# convert to DGE list object and apply Normalization
dge <- DGEList(expr.matrix.keep)

dge.norm <- calcNormFactors(dge) # using default settings
head(dge.norm$counts) %>% kable()

# log2 transform and check results
dge.log <- cpm(y = dge.norm, log = TRUE, normalized.lib.sizes = FALSE, lib.size = NA) # use the cpm function to log transform, but set other parameters to FALSE/NA to avoid taking "per million" since my values are already in TPM

head(dge.log) %>% kable()
head(log2(dge.norm$counts + 0.25)) %>% kable() # reasonably similar to the result obtained with cpm function, which uses prior.count = 0.25 as default
```


## Perform linear modeling and statistics
```{r limma_stats}
myfit <- lmFit(dge.log, design = designMatrix)
myfitEb <- eBayes(myfit, trend = TRUE) # trend analysis for RNA-seq data


# make contrast table for desired comparisons (each vs. ctrl)
myContrasts <- makeContrasts(m143_vs_ctrl = SPG_targetmiR143 - Intercept,
                             mX_vs_ctrl = SPG_targetmiRX - Intercept,
                             both_vs_ctrl = SPG_targetboth - Intercept,
                             none_vs_ctrl = SPG_targetnone - Intercept, levels = designMatrix)

# re-fit model with contrasts
contfit <- contrasts.fit(myfit, myContrasts) # I double checked and the names do match; the STAT540 seminar-4 tutorial also threw the same error message
contfitEb <- eBayes(contfit, trend = TRUE)

contResult <- topTable(contfitEb, number = Inf)

head(contResult) %>% kable()


```

### QC: p-value histogram
```{r}
hist(contResult$P.Value)
```

> NOTE: Too many low p-values???


## Identify differentially expressed genes in contrasts
```{r}
testResults <- decideTests(contfitEb, p.value = 1e-4, p.adjust.methods = "fdr", method = "global")

summary(testResults)
```

> NOTE: A lot of genes are massively downregulated... there must be something wrong!



# DE-seq  (DO NOT USE!)

> As I determined in the FBXO11 expression analaysis that Limma analysis is better powered to analyzing RNA-seq data with multiple gorups, I performed Limma instead (added above) and will proceed with those results.

```{r make_deseq}
library(DESeq2)
design.table <- select(libs, library_name, SPG_target) %>%
  column_to_rownames(var = "library_name")

expr.matrix <- column_to_rownames(expr, var = "Name") %>%
  select(rownames(design.table)) %>% # put columns in same order as the design matrix
  as.matrix()

# filter to "highly expressed" genes
thresh <- expr.matrix >= 1
keep <- rowSums(thresh) >= 2 # keep rows where there are > 2 samples with tpm > 1

expr.matrix.keep <- expr.matrix[keep,]

# convert to integers for DEseq
int.matrix.keep <- round(expr.matrix.keep, digits = 0)

dds <- DESeqDataSetFromMatrix(countData = int.matrix.keep, colData = design.table, design = ~SPG_target)

res <- DESeq(dds)
```


### Summarise results by each SPG group  

#### miR-143 vs. ctrl  
```{r results_143}
resultsNames(res)

result.pair <- results(res, contrast = c("SPG_target", "miR.143", "ctrl")) # contrast: the factor, then numerator, then denominator

kable(head(result.pair[which(result.pair$padj < 0.05),]))

write.csv(x = result.pair, file = "./Kate/DEseq/miR143_vs_ctrl.csv")
```

```{r miR143_volcano}
# cutoffs: padj < 0.05, |FC| > 2
dat <- read.csv("./Kate/DEseq/miR143_vs_ctrl.csv") %>%
  mutate(Signif_p = -log10(padj) > -log10(0.05), 
         Signif_delta = abs(log2FoldChange) > log2(2),
         Signif = Signif_p == TRUE & Signif_delta == TRUE)

ggplot(dat, aes(log2FoldChange, -log10(padj))) +
  geom_point(aes(colour = Signif))
```



#### miR-X vs ctrl  
```{r results_X}
result.pair <- results(res, contrast = c("SPG_target", "miR.X", "ctrl")) # contrast: the factor, then numerator, then denominator

kable(head(result.pair[which(result.pair$padj < 0.05),]))

write.csv(x = result.pair, file = "./Kate/DEseq/miRX_vs_ctrl.csv")
```

```{r miRX_volcano}
# cutoffs: padj < 0.05, |FC| > 2
dat <- read.csv("./Kate/DEseq/miRX_vs_ctrl.csv") %>%
  mutate(Signif_p = -log10(padj) > -log10(0.05), 
         Signif_delta = abs(log2FoldChange) > log2(2),
         Signif = Signif_p == TRUE & Signif_delta == TRUE)

ggplot(dat, aes(log2FoldChange, -log10(padj))) +
  geom_point(aes(colour = Signif))
```


#### Both vs ctrl  
```{r results_original_spg}
result.pair <- results(res, contrast = c("SPG_target", "both", "ctrl")) # contrast: the factor, then numerator, then denominator

kable(head(result.pair, 20)) # all p-values = 1, why??? Maybe the SPG didn't work to knock anything down? Maybe there was an outlier sample? ASK

write.csv(x = result.pair, file = "./Kate/DEseq/originalSPG_vs_ctrl.csv")
```

```{r orig_spg_volcano}
# cutoffs: padj < 0.05, |FC| > 2
dat <- read.csv("./Kate/DEseq/originalSPG_vs_ctrl.csv") %>%
  mutate(Signif_p = -log10(padj) > -log10(0.05), 
         Signif_delta = abs(log2FoldChange) > log2(2),
         Signif = Signif_p == TRUE & Signif_delta == TRUE)

ggplot(dat, aes(log2FoldChange, -log10(padj))) +
  geom_point(aes(colour = Signif))
```

