---
title: "Expression Analysis of TIRAP-oe vs. MIG-ctrl"
author: "Jennifer Grants"
date: "2/6/2019"
output: 
  html_document:
    keep_md: true
    toc: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, eval = TRUE)

library(tidyverse)
library(limma)
library(edgeR)
library(pheatmap)
library(RColorBrewer)
library(knitr)
```


# Expression data and design
```{r}
load("./TIRAP/expr.rda")
load("./TIRAP/libs.rda")

head(expr) %>% kable()
head(libs) %>% kable()
```

## QC plot
```{r qc_expression_hist}
column_to_rownames(expr, var = "gene_id") %>%
  as.matrix() %>%
  log2() %>%
  hist(main = "log2(Expression)")
```


## Processing expression data
```{r}
expr.matrix <- column_to_rownames(expr, var = "gene_id") %>%
  as.matrix()

# filter to 'higher' expressed genes (RPKM >= 0.1 in at least 3 samples)
thresh <- expr.matrix >= 0.1
keep <- rowSums(thresh) >= 3

expr.matrix.keep <- expr.matrix[keep,]

# normalization
dge <- DGEList(expr.matrix.keep)

dge.norm <- calcNormFactors(dge)

# log2 transformation
dge.log <- cpm(dge.norm, normalized.lib.sizes = FALSE, lib.size = NA, log = TRUE, prior.count = 0.01) # prior.count changes the addition term for log2 transformation

head(dge.log) %>% kable()
```

## Make design matrix
```{r}
# First create a factor for experimental design
design <- select(libs, library_name, specimen_subset_external_id) %>%
  rename(specimen = specimen_subset_external_id) %>%
  mutate(oe = specimen) %>%
  separate(col = oe, into = "construct", remove = T)

# Model matrix
designMatrix <- model.matrix(~construct, design)

kable(designMatrix)
```


# Limma analysis
## Top table
```{r}
fit <- lmFit(dge.log, design = designMatrix)
fitEb <- eBayes(fit)

result <- topTable(fitEb, number = Inf)

head(result, 10)
```

```{r}
write.csv(result, "./TIRAP/Limma/Limma_result_TIRAPvsMIG.csv")
```

### QC: p-value distribution
```{r QC_pval_distr}
hist(result$P.Value, main = "Distribution of p-values")
```


## Heatmap of significantly differentially expressed genes
```{r}
cutoff <- 0.05

signif <- decideTests(fitEb, p.value = cutoff, p.adjust.methods = "fdr", method = "global")
summary(signif)

hits <- as.data.frame(signif) %>%
  rownames_to_column(var = "gene") %>%
  filter(constructTIRAP != 0)

hits.log.expr <- dge.log[which(rownames(dge.log) %in% hits$gene),]

hits.scaled <- t(hits.log.expr) %>% scale() %>% (t)
```

```{r heatmap_DE_genes}
heatmap_pallete <- colorRampPalette(brewer.pal(8, name = "RdBu"))(21) %>% rev

anno.frame <- data.frame(sample.name = colnames(hits.scaled), Construct = design[match(colnames(hits.scaled), design$library_name),]$construct) %>%
  column_to_rownames(var = "sample.name")

pheatmap(hits.scaled, 
         cluster_cols = F, 
         cluster_rows = T, 
         show_rownames = F, 
         clustering_method = "ward.D2", 
         clustering_distance_cols = "euclidean", 
         annotation_col = anno.frame, 
         color = heatmap_pallete)
```

## Heatmap of top 50 up and top 50 down genes with names

Prep for making heatmap with gene names:  
```{r}
# top upregulated
up <- as.data.frame(signif) %>%
  rownames_to_column(var = "gene") %>%
  filter(constructTIRAP > 0)

up_result <- result[which(rownames(result) %in% up$gene),] %>%
  rownames_to_column(var = "gene") %>%
  arrange(adj.P.Val) %>%
  head(50)

up.log.expr <- dge.log[which(rownames(dge.log) %in% up_result$gene),]
up.scaled <- t(up.log.expr) %>% scale() %>% (t)

# top downregulated
down <- as.data.frame(signif) %>%
  rownames_to_column(var = "gene") %>%
  filter(constructTIRAP < 0)

down_result <- result[which(rownames(result) %in% down$gene),] %>%
  rownames_to_column(var = "gene") %>%
  arrange(adj.P.Val) %>%
  head(50)

down.log.expr <- dge.log[which(rownames(dge.log) %in% down_result$gene),]
down.scaled <- t(down.log.expr) %>% scale() %>% (t)

# biomart gene name conversion to mgi symbol
library(biomaRt)

mouse <- useMart(biomart = "ENSEMBL_MART_ENSEMBL", dataset = "mmusculus_gene_ensembl")
genenames_up <- getBM(attributes = c("ensembl_gene_id", "mgi_symbol"), filters = "ensembl_gene_id", values = rownames(up.scaled), mart = mouse)
genenames_down <- getBM(attributes = c("ensembl_gene_id", "mgi_symbol"), filters = "ensembl_gene_id", values = rownames(down.scaled), mart = mouse)

up.scaled <- as.data.frame(up.scaled) %>%
  rownames_to_column(var = "gene_id")
down.scaled <- as.data.frame(down.scaled) %>%
  rownames_to_column(var = "gene_id")

up.scaled$gene <- genenames_up[match(up.scaled$gene_id, genenames_up$ensembl_gene_id),]$mgi_symbol
down.scaled$gene <- genenames_down[match(down.scaled$gene_id, genenames_down$ensembl_gene_id),]$mgi_symbol

up.scaled <- dplyr::select(up.scaled, -gene_id) %>%
  column_to_rownames(var = "gene")

down.scaled <- dplyr::select(down.scaled, -gene_id) %>%
  filter(!is.na(gene)) %>%
  column_to_rownames(var = "gene")

scaled.50.50 <- rbind(up.scaled, down.scaled)
```

Draw heatmap:  
```{r heatmap_top50_up_down, fig.height=16, fig.width = 8}
pheatmap(scaled.50.50, 
         cluster_cols = F, 
         cluster_rows = T, 
         show_rownames = T, 
         clustering_method = "ward.D2", 
         clustering_distance_cols = "euclidean", 
         annotation_col = anno.frame, 
         color = heatmap_pallete)
```



# Prepare data for GSEA analysis (expression analysis)
## Convert to human gene names  
```{r}
human <- useMart("ENSEMBL_MART_ENSEMBL", "hsapiens_gene_ensembl")

orthologs <- getBM(attributes = c("ensembl_gene_id", "hsapiens_homolog_ensembl_gene"), filters = "ensembl_gene_id", values = rownames(expr.matrix.keep), mart = mouse) # using "higher" expressed genes only, in expr.matrix.keep

hgnc <- getBM(attributes = c("ensembl_gene_id", "hgnc_symbol"), filters = "ensembl_gene_id", values = orthologs$hsapiens_homolog_ensembl_gene, mart = human)

# Add human gene IDs and names to expr.matrix.keep (using "higher" expressed genes only to avoid including a whole bunch of crap)
expr.matrix.keep <- as.data.frame(expr.matrix.keep) %>%
  rownames_to_column(var = "gene_id")
expr.matrix.keep$human_gene_id <- orthologs[match(expr.matrix.keep$gene_id, orthologs$ensembl_gene_id),]$hsapiens_homolog_ensembl_gene
expr.matrix.keep$hgnc_name <- hgnc[match(expr.matrix.keep$human_gene_id, hgnc$ensembl_gene_id),]$hgnc_symbol
  # also add mouse names to double check fidelity of conversion to human
mgi <- getBM(attributes = c("ensembl_gene_id", "mgi_symbol"), filters = "ensembl_gene_id", values = expr.matrix.keep$gene_id, mart = mouse)
expr.matrix.keep$mgi_name <- mgi[match(expr.matrix.keep$gene_id, mgi$ensembl_gene_id),]$mgi_symbol # looks good

head(expr.matrix.keep) %>% kable()
```

## Make expression dataset
```{r}
eset.keep <- mutate(expr.matrix.keep, DESCRIPTION = "na") %>%
  rename(NAME = hgnc_name) %>%
  dplyr::select(NAME, DESCRIPTION, contains("A54")) %>%
  filter(!is.na(NAME), NAME != "")

# write to .txt with tab sep for GSEA
write.table(x = eset.keep, file = "./TIRAP/GSEA_fromLimma/TIRAP_exprSet_filteredForHighExpression.txt", sep = "\t", row.names = F, quote = F)
```


## Make .cls file
```{r}
# check that sample names are in same order on design frame and eset.keep
identical(colnames(eset.keep[,3:8]), design$library_name, attrib.as.set = T)

# .cls file rows
numbers <- c(nrow(design), length(unique(design$construct)), 1) # 1st line of .cls file: samples, groups, (always 1)

names <- c("#", unique(design$construct)) %>% unlist() # 2nd line of .cls file: #symbol + names of groups

groups <- unlist(design$construct) # 3rd line of .cls file: group assignment of each sample (in the same order as colnames in exprset)

# combine into 1 .cls file
write(x = numbers, file = "./TIRAP/GSEA_fromLimma/Phenotypes_TIRAP.cls", ncolumns = length(numbers))
write(x = names, file = "./TIRAP/GSEA_fromLimma/Phenotypes_TIRAP.cls", ncolumns = length(names), append = TRUE)
write(x = groups, file = "./TIRAP/GSEA_fromLimma/Phenotypes_TIRAP.cls", ncolumns = length(groups), append = TRUE)
```

> Ready to do GSEA and EnrichmentMap analysis






